File: fackcar/src/main/java/netposa/fakecar/feature/rpc/ResultRecord.java
package netposa.fakecar.feature.rpc;
/**
 * 向数据库输出结果的类定义
 * @author hongs.yang
 *
 */
public class ResultRecord {
	
	private final KeyRecord key;
	private final ValueRecord firstValue;
	private final ValueRecord secondValue;
	
	public ResultRecord(ValueRecord firstValue) {
		this.key = null;
		this.firstValue = firstValue;
		this.secondValue = null;
	}
	
	public ResultRecord(KeyRecord key, 
			ValueRecord firstValue, ValueRecord secondValue) {
		this.key = key;
		this.firstValue = firstValue;
		this.secondValue = secondValue;
	}


	public KeyRecord getKey() {
		return key;
	}
	public ValueRecord getFirstValue() {
		return firstValue;
	}
	public ValueRecord getSecondValue() {
		return secondValue;
	}
	
	

}

File: fackcar_spark/src/main/java/com/netposa/poseidon/fakecardetect/bean/ResultRecord.scala
package com.netposa.poseidon.fakecardetect.bean

class ResultRecord(inKey:String,inFirstValue:ValueRecord,inSecondValue:ValueRecord,inFlag:Boolean) extends Serializable{

  private var key:String = inKey
  private var firstValue:ValueRecord = inFirstValue
  private var secondValue:ValueRecord = inSecondValue
  private var flag:Boolean = inFlag//true表示是套牌车需要入库，False表示不是套牌需要更新全局信息
  def this(inFirstValue:ValueRecord,inFlag:Boolean){
    this(null,inFirstValue,null,inFlag)
  }
  def this(inKey:String,inFirstValue:ValueRecord,inFlag:Boolean){
    this(inKey,inFirstValue,null,inFlag)
  }
  def getKey():String={
    return key
  }
  def getFirstValue():ValueRecord = {
    return firstValue
  }
  def getSecondValue():ValueRecord ={
    return secondValue
  }
  def getFlag():Boolean={
    return flag
  }



}
